_studio/shared/umc/codec/h265_dec/src/umc_h265_bitstream_headers.cpp

```diff
@@ -1340,7 +1341,16 @@ void H265HeadersBitstream::xParsePredWeightTable(const H265SeqParamSet *sps, H26
             wp[1].log2_weight_denom = sliceHdr->chroma_log2_weight_denom;
             wp[2].log2_weight_denom = sliceHdr->chroma_log2_weight_denom;
 
-            wp[0].present_flag = Get1Bit(); // luma_weight_lX_flag
+            if (sliceHdr->m_RefPOCList[eRefPicList][iRefIdx] == sliceHdr->m_poc)
+            {
+                iPresent = 0;
+            }
+            else
+            {
+                iPresent = Get1Bit();
+            }
+
+            wp[0].present_flag = (iPresent == 1); // luma_weight_lX_flag
```

Comment:
why do you need to compare with one, if you acquiring just one bit?

I believe following code should work
```
            if (sliceHdr->m_RefPOCList[eRefPicList][iRefIdx] == sliceHdr->m_poc)
            {
                wp[0].present_flag = 0; // luma_weight_lX_flag
            }
            else
            {
                wp[0].present_flag = Get1Bit(); // luma_weight_lX_flag
            }
```

Reply:
Good suggestion! Code is updated.

Reply:
@mgonchar , +1 from you?

Reply:
@dmitryermilov looks fine to me, but I'm not familiar with this code